core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -605,6 +603,29 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  def unZip(inFile: File, unzipDir: File): Unit = {
+    if (!unzipDir.mkdirs && !unzipDir.isDirectory) {
+      throw new IOException("Mkdirs failed to create " + unzipDir)
+    } else {
+      if (Shell.WINDOWS) {
+        FileUtil.unZip(inFile, unzipDir)
+      } else {
+        val untarCommand = new StringBuilder
+        untarCommand.append(s"mkdir $unzipDir && unzip -d ")
+        untarCommand.append(FileUtil.makeSecureShellPath(unzipDir))
+        untarCommand.append(" -o ")
+        untarCommand.append(FileUtil.makeSecureShellPath(inFile))
+        val shellCmd = Array("bash", "-c", untarCommand.toString)
+        val shexec = new ShellCommandExecutor(shellCmd)
+        shexec.execute()
+        val exitcode = shexec.getExitCode
+        if (exitcode != 0) {
+          throw new IOException("Error untarring file " + inFile + ". Tar process exited with exit code " + exitcode)
```

Comment:
- unzip
- Just suggestion that it's better with shell errorMsg

Reply:
Thank you for your advice. This is consistent with the exception thrown by untar decompression. But I do need to change the command name.